packages/medusa/src/services/swap.ts

```diff
@@ -350,24 +380,28 @@ class SwapService extends BaseService {
         }
       }
 
-      const newItems = await Promise.all(
-        additionalItems.map(({ variant_id, quantity }) => {
-          return this.lineItemService_.generate(
-            variant_id,
-            order.region_id,
-            quantity
-          )
-        })
-      )
+      let newItems: LineItem[] = []
+
+      if (additionalItems) {
+        newItems = await Promise.all(
+          additionalItems.map(({ variant_id, quantity }) => {
+            return this.lineItemService_.generate(
+              variant_id,
+              order.region_id,
+              quantity
+            )
+          })
+        )
+      }
 
       const evaluatedNoNotification =
         no_notification !== undefined ? no_notification : order.no_notification
```

Comment:
here is the logic that is used to infer the value of `no_notification` - see previous comment in create-swap-shipment.